staging/src/k8s.io/apiserver/pkg/storage/etcd3/errors.go

```diff
@@ -30,13 +32,26 @@ func interpretWatchError(err error) error {
 	return err
 }
 
-func interpretListError(err error, paging bool) error {
-	switch {
-	case err == etcdrpc.ErrCompacted:
-		if paging {
-			return errors.NewResourceExpired("The provided from parameter is too old to display a consistent list result. You must start a new list without the from.")
-		}
+func interpretListError(err error, continueKey, keyPrefix string) error {
+	if err != etcdrpc.ErrCompacted {
+		return err
+	}
+	if len(continueKey) == 0 {
 		return errors.NewResourceExpired("The resourceVersion for the provided list is too old.")
 	}
+	message := "The provided from parameter is too old to display a consistent list result. " +
+		"You can start a new list without the from."
+	// continueToken.ResoureVersion=-1 means that the apiserver can
+	// continue the list at the latest resource version. We don't use rv=0
+	// for this purpose to distinguish from a bad token that has empty rv.
+	newToken, err := encodeContinue(continueKey, keyPrefix, -1)
+	if err != nil {
+		return errors.NewResourceExpired(message +
```

Comment:
This could leak database state information to the caller.  You should probably be using utilruntime.HandleError and return a generic error message.

Reply:
Done, but we are returning `encodeContinue` err in other place: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L623.